src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs

```diff
@@ -145,6 +145,7 @@ internal UnitTestResult[] Execute()
 
             if (isIgnoreAttributeOnClass || isIgnoreAttributeOnMethod)
             {
+                this.testMethodInfo.TestExecutionRecorder.RecordStartAndEnd(new UnitTestElement(this.test), UnitTestOutcome.Ignored);
```

Comment:
Why do we need to log these ?

Reply:
@singhsarab strictly speaking logging the start/end events for these error cases probably isn't useful but I was trying to keep parity with the existing logic I'm replacing. One thing I should check is if VSTEST will log the result if no start/end event happens (I think it will though). If you are fine with the slight behavior change then I can remove these.

Reply:
@singhsarab let me know what you think about this. I've addressed all other comments.

Reply:
@NGloreous : Can you explain a bit about keeping parity with the existing logic? I don't think recording testStartAndEnd for such error/exception cases should be required, since the wrapper over this class i.e. TestExecutionManager.cs is already taking care of handling Start and End.

Reply:
@jayaranigarg TestExecutionManager.cs no longer fires start/end events which is the whole point of my change (to move the start/end event firing closer to where the test actually executes so we capture these events for each data driven iteration).

Reply:
@NGloreous : In the process of removing start/end events from TestExecutionManager.cs, we might end up missing certain scenarios.
To have start/end events for data driven testing, I would recommend adding events in TestMethdRunner.cs [here](https://github.com/microsoft/testfx/blob/master/src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs#L315) and [here](https://github.com/microsoft/testfx/blob/master/src/Adapter/MSTest.CoreAdapter/Execution/TestMethodRunner.cs#L257). I would let TestExecutionManager.cs still fire start/end events which can act as parent results and firing from TestMethodRunner.cs can act as child results.

Reply:
@jayaranigarg letting TestExecutionManager.cs still fire start/end events for the parent result seems reasonable. Firing events for data driven tests at the location you suggest doesn't solve for the case that a custom test execution extension runs a test multiple times (we plan to leverage this extension to do test retry and want a start/end event for each try). What are your concerns with the current placement of start/end event firing inside TestMethodInfo.cs? Please advise and I will update the PR.

Reply:
@NGloreous : Good catch about custom test execution extension. In that case, let's keep the start/end event firing inside TestMethodInfo.cs itself.
Let TestExecutionManager.cs still fire start and end events for parent result, which will also cover any error scenarios. So we can safely remove any event firing from TestMethodRunner.cs.

Reply:
Will do. Thanks for the feedback. I'll wok on this later in the week and update the PR.